lte/gateway/c/core/oai/tasks/amf/amf_app_handler.cpp

```diff
@@ -629,16 +633,19 @@ int amf_app_handle_pdu_session_response(
         ue_context->mm_state = REGISTERED_CONNECTED;
       }
     } else {
-      OAILOG_DEBUG(LOG_AMF_APP,
-                   "Sending message to gNB for PDUSessionResourceSetupRequest "
-                   "**n_active_pdus=%d **\n",
-                   smf_ctx->n_active_pdus);
-
       amf_smf_msg.pdu_session_id = pdu_session_resp->pdu_session_id;
       /*Execute PDU establishement accept from AMF to gnodeb */
-      pdu_state_handle_message(
-          REGISTERED_CONNECTED, STATE_PDU_SESSION_ESTABLISHMENT_ACCEPT,
-          CREATING, ue_context, amf_smf_msg, NULL, pdu_session_resp, ue_id);
+      if (smf_ctx->pdu_session_state == CREATING) {
+        event = STATE_PDU_SESSION_ESTABLISHMENT_ACCEPT;
+      } else if (smf_ctx->pdu_session_state == ACTIVE) {
+        event = STATE_PDU_SESSION_MODIFICATION_REQUEST;
+      } else if (smf_ctx->pdu_session_state == PENDING_RELEASE) {
+        event = STATE_PDU_SESSION_RELEASE_COMPLETE;
+      }
+
+      pdu_state_handle_message(REGISTERED_CONNECTED, event,
```

Comment:
On failing to send PDU session modification request to gNB; what action is taken towards SMF?